routing/missioncontrol.go

```diff
@@ -299,6 +258,30 @@ func (m *MissionControl) getEdgeProbabilityForNode(nodeHistory *nodeHistory,
 	return probability
 }
 
+// getPairProbability estimates the probability of successfully
+// traversing from fromNode to toNode based on historical payment outcomes.
+func (m *MissionControl) getPairProbability(fromNode,
+	toNode route.Vertex, amt lnwire.MilliSatoshi) float64 {
+
+	// Start by getting the last node level failure. If there is none,
+	// lastFail will be zero.
+	lastFail := m.lastNodeFailure[fromNode]
+
+	// Retrieve the last pair outcome.
+	pair := NewDirectedNodePair(fromNode, toNode)
+	lastPairResult, ok := m.lastPairFailure[pair]
+
+	// If there is none or it happened before the last node level failure,
+	// the node level failure is the most recent and thus returned.
+	if ok && lastPairResult.timestamp.After(lastFail) {
+		if amt >= lastPairResult.minPenalizeAmt {
```

Comment:
Shouldn't this be the other way around? Only if the amount were sending is less than the amount for the last failure should we use this failure timestamp.

Reply:
Using this failures timestamp means that we do apply the penalty based on the previous attempt. We'd only want to do that if the amount is greater or equal than the previous attempt amount. So smaller amounts get a higher probability, because it could work for those.